tests/functional/modules/test_zos_tso_command_func.py

```diff
@@ -45,89 +47,53 @@ def test_zos_tso_command_long_command_128_chars(ansible_zos_module):
-def test_zos_tso_command_auth_command_listds_using_alias(ansible_zos_module):
-    hosts = ansible_zos_module
-    results = hosts.all.zos_tso_command(command=["LISTDS 'imstestl.ims1.temp.ps'"])
+    # Validate listds auth of a dataset with commands
+    results = hosts.all.zos_tso_command(commands=["LISTDS '{0}'".format(DEFAULT_TEMP_DATASET)])
     for result in results.contacted.values():
-        assert result.get("output")[0].get("rc") == 0
+        for item in result.get("output"):
+            assert item.get("rc") == 0
         assert result.get("changed") is True
-
-
-# The positive path test
-# tests that alias "command" works
-def test_zos_tso_single_command_auth_command_listds_using_alias(ansible_zos_module):
-    hosts = ansible_zos_module
-    results = hosts.all.zos_tso_command(command="LISTDS 'imstestl.ims1.temp.ps'")
+    # Validate listds auth single command
+    results = hosts.all.zos_tso_command(command="LISTDS '{0}'".format(DEFAULT_TEMP_DATASET))
     for result in results.contacted.values():
-        assert result.get("output")[0].get("rc") == 0
+        for item in result.get("output"):
+            assert item.get("rc") == 0
         assert result.get("changed") is True
-
-
-# The positive path test
-def test_zos_tso_command_unauth_command_listcat(ansible_zos_module):
-    hosts = ansible_zos_module
+    # Validate LISTCAT comand
     results = hosts.all.zos_tso_command(
-        commands=["LISTCAT ENT('imstestl.ims1.temp.ps')"]
+        commands=["LISTCAT ENT('{0}')".format(DEFAULT_TEMP_DATASET)]
     )
     for result in results.contacted.values():
-        assert result.get("output")[0].get("rc") == 0
+        for item in result.get("output"):
+            assert item.get("rc") == 0
         assert result.get("changed") is True
-
-
-# The positive path test
-def test_zos_tso_command_both_unauth_and_auth_command(ansible_zos_module):
-    hosts = ansible_zos_module
-    results = hosts.all.zos_tso_command(commands=["delete 'imstestl.ims1.temp.ps'"])
+    # Validate remove dataset
+    results = hosts.all.zos_tso_command(commands=["delete '{0}'".format(DEFAULT_TEMP_DATASET)])
     for result in results.contacted.values():
-        assert result.get("output")[0].get("rc") == 0
+        for item in result.get("output"):
+            assert item.get("rc") == 0
         assert result.get("changed") is True
-
-
-# The failure path test
-# the dataset has be deleted.
-def test_zos_tso_command_valid_command_failed_as_has_been_deleted(ansible_zos_module):
-    hosts = ansible_zos_module
-    results = hosts.all.zos_tso_command(commands=["delete 'imstestl.ims1.temp.ps'"])
+    # Validate already was remove
+    results = hosts.all.zos_tso_command(commands=["delete '{0}'".format(DEFAULT_TEMP_DATASET)])
     for result in results.contacted.values():
-        assert result.get("output")[0].get("rc") == 8
+        for item in result.get("output"):
+            assert item.get("rc") == 8
```

Comment:
A little nit-picking here, I think it would be beneficial to add a comment explaining why the expected rc is 8. Maybe something like:

"# Expect the tso_command to fail here because the previous command will have already deleted the data set"